editoast/src/views/authz.rs

```diff
@@ -145,6 +152,126 @@ pub(in crate::views) async fn user_groups(
     Ok(Json(result))
 }
 
+#[derive(Deserialize, Clone, ToSchema)]
+pub struct UsersInfoRequest {
+    #[serde(default)]
+    ids: Vec<i64>,
+    #[serde(default)]
+    identities: Vec<String>,
+}
+
+#[derive(Serialize, ToSchema)]
+pub struct UserInfo {
+    id: i64,
+    name: String,
+    identities: Vec<String>,
+    roles: HashSet<Role>,
+    #[schema(inline)]
+    groups: HashSet<Group>,
+}
+
+#[editoast_derive::route]
+#[utoipa::path(
+    post,
+    path = "",
+    tag = "authz",
+    request_body(
+        content = inline(UsersInfoRequest),
+        description = "A list of user IDs and identities to get information on",
+    ),
+    responses((
+        status = 200,
+        description = "Get information on a list of users",
+        body = inline(Vec<UserInfo>),
+    ))
+)]
+pub(in crate::views) async fn users_info(
+    Extension(auth): AuthenticationExt,
+    State(AppState {
+        regulator, db_pool, ..
+    }): State<AppState>,
+    Json(body): Json<UsersInfoRequest>,
+) -> Result<Json<Vec<UserInfo>>> {
+    let authorized = auth
+        .check_roles([Role::Admin].into())
+        .await
+        .map_err(AuthorizationError::AuthError)?;
+    if !authorized {
+        return Err(AuthorizationError::Forbidden.into());
+    }
+    let UsersInfoRequest {
+        ids: user_ids,
+        identities: user_identities,
+    } = body.clone();
+
+    // Retrieve users by IDs
+    let settings = SelectionSettings::new().filter(move || User::ID.eq_any(user_ids.clone()));
+    let users_1 = User::list(&mut db_pool.get().await?, settings).await?;
+    if users_1.len() < body.ids.len() {
+        let found_ids: HashSet<_> = users_1.iter().map(|u| u.id).collect();
+        let missing_id = body.ids.into_iter().find(|id| !found_ids.contains(id));
+        return Err(AuthzError::UnknownUser {
+            id: missing_id.unwrap(),
+        }
+        .into());
+    }
```

Comment:
```suggestion
    let users_1 = User::retrieve_batch_or_fail(&mut db_pool.get().await?, |missing| { /* handle missing users */).await?;
```

I think this would be the same?